torchaudio/__init__.py

```diff
@@ -122,23 +124,21 @@ def save(filepath, src, sample_rate, precision=16, channels_first=True):
 
 
 @_audio_backend_guard("sox")
-def save_encinfo(filepath,
-                 src,
-                 channels_first=True,
-                 signalinfo=None,
-                 encodinginfo=None,
-                 filetype=None):
+def save_encinfo(filepath: str,
+                 src: Tensor,
+                 signalinfo: SignalInfo,
+                 encodinginfo: EncodingInfo,
+                 channels_first: bool = True,
+                 filetype: Optional[str] = None) -> None:
```

Comment:
Let's not make a BC-breaking change in this PR. Can you clarify what issue you are trying to fix?

As used [here](https://github.com/pytorch/audio/blob/master/torchaudio/_sox_backend.py#L65), `encodinginfo` is indeed optional.

Reply:
Sure :) 
 in line 168 we have:
```
    if not isinstance(signalinfo.rate, float):
        if float(signalinfo.rate) == signalinfo.rate:
            signalinfo.rate = float(signalinfo.rate)
        else:
            raise TypeError('Sample rate should be a float or int')
    # check if the bit precision (i.e. bits per sample) is an integer
    if not isinstance(signalinfo.precision, int):
        if int(signalinfo.precision) == signalinfo.precision:
            signalinfo.precision = int(signalinfo.precision)
        else:
            raise TypeError('Bit precision should be an integer')
```
however if `signalinfo` is `None` we cannot get `rate` so will break, the solution that I first work was using an `if signalinfo:` and indent the rest (lines 168-178), which would only do those checks if the parameter was actually defined, however when I tried to run `_torch_sox.write_audio_file` and if you look into its typing is mandatory to have `signalinfo` defined, so I opted by changing it to mandatory, however, it can be fixed by using the if-condition and maybe adding a raise if its not defined. Your call :)

Reply:
@vincentqb Updated to not break BC